src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1803,39 +1832,49 @@ public override BoundNode VisitConversion(BoundConversion node)
                     case ConversionKind.ImplicitUserDefined:
                         if ((object)node.SymbolOpt != null && node.SymbolOpt.ParameterCount == 1)
                         {
-                            WarnOnNullReferenceArgument(operand, _result.Type, node.SymbolOpt.Parameters[0], expanded: false);
+                            WarnOnNullReferenceArgument(operand, operandType, node.SymbolOpt.Parameters[0], expanded: false);
                         }
                         break;
 
                     case ConversionKind.AnonymousFunction:
                         if (!node.ExplicitCastInCode && operand.Kind == BoundKind.Lambda)
                         {
                             var lambda = (BoundLambda)operand;
-                            ReportNullabilityMismatchWithTargetDelegate(operand.Syntax, node.Type.GetDelegateType(), lambda.Symbol);
+                            ReportNullabilityMismatchWithTargetDelegate(operand.Syntax, targetType.GetDelegateType(), lambda.Symbol);
                         }
                         break;
 
                     case ConversionKind.MethodGroup:
                         if (!node.ExplicitCastInCode)
                         {
-                            ReportNullabilityMismatchWithTargetDelegate(operand.Syntax, node.Type.GetDelegateType(), node.SymbolOpt);
+                            ReportNullabilityMismatchWithTargetDelegate(operand.Syntax, targetType.GetDelegateType(), node.SymbolOpt);
                         }
                         break;
                 }
 
-                var operandType = _result.Type;
+                if (node.ExplicitCastInCode &&
+                    node.IsNullable == false &&
+                    targetType.IsReferenceType &&
+                    (operandType?.IsNullable == true || (operandType is null && operand.IsLiteralNullOrDefault())))
+                {
+                    // PROTOTYPE(NullableReferenceTypes): Should not report warning for explicit
+                    // user -defined conversion if the operator is defined from nullable to
+                    // non-nullable. See StaticNullChecking.ExplicitCast_UserDefined.
+                    ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_ConvertingNullableToNonNullable, node.Syntax);
```

Comment:
>node [](start = 103, length = 4)

operand?

Reply:
The warning is for the conversion rather than the operand. This matches the span used for other explicit conversion errors such as `ERR_NoExplicitConv` and `ERR_ConvertToStaticClass`.

---
In reply to: [175656212](https://github.com/dotnet/roslyn/pull/25516#discussion_r175656212) [](ancestors = 175656212)